feat(cli): check if dotfiles install script is executable (#8588)
coder/coder: docs/dotfiles.md

# Install some of my favorite tools every time my workspace boots
sudo apt install -y neovim fish cargo
```

> ## Install or Setup script support

Reviewer: For simplicity, let's pick one so we can refer to it in the text below:

```suggestion
## Setup script support
```

> User can have either of the following setup script files in their dotfiles repo:

Reviewer: ```suggestion
User can setup their dotfiles by creating one of the following script files in their dotfiles repo:
```

> - install.sh
> - install
> - bootstrap.sh
> - bootstrap
> - script/bootstrap
> - setup.sh
> - setup
> - script/setup

Reviewer: ```suggestion
- `install.sh`
- `install`
- `bootstrap.sh`
- `bootstrap`
- `script/bootstrap`
- `setup.sh`
- `setup`
- `script/setup`
```

> If any of the above files are found in the specified order, Coder will execute the first match, and the rest will be skipped.

Reviewer: ```suggestion
If one the above files is found (in the specified order), Coder will try to execute the first match. After the first match is found, other files will be ignored.
```

> If any of these files are present, Coder will run them after cloning the dotfiles repo. If any of these files are found in the specified order, only the first match will be executed, and the rest will be skipped.

Reviewer 2: ```suggestion
If any of the above files are found in the specified order, Coder will execute the first match, and the rest will be skipped.
```

> The Setup scripts must be executable. If they are not, Coder will not execute them. If the script is not executable, you can make it executable by running:

Reviewer: ```suggestion
The setup script must be executable, otherwise the dotfiles setup will fail. If you encounter this issue, you can fix it by making the script executable using the following commands:
```

> ```bash
> cd <path_to_dotfiles_repo>
> 
> chmod +x <script_name>
> 
> git commit -m "Make <script_name> executable" <script_name>
> 
> git push

Reviewer: ```suggestion
```shell
cd <path_to_dotfiles_repo>
chmod +x <script_name>
git commit -m "Make <script_name> executable" <script_name>
git push
```
